core/types/gen_tx_json.go

```diff
@@ -20,7 +20,7 @@ func (t txdata) MarshalJSON() ([]byte, error) {
 		Price        *hexutil.Big    `json:"gasPrice"   gencodec:"required"`
 		GasLimit     hexutil.Uint64  `json:"gas"        gencodec:"required"`
 		ShardID      uint32          `json:"shardID"    gencodec:"required"`
-		ToShardID    uint32          `json:"toShardID"`
+		ToShardID    uint32          `json:"toShardID"  gencodec:"required"`
```

Comment:
Looks like this file was desynced from the source file (changed at commit 02fb5cb7b2de48f546281332c3b83a251d0507d7 on 8/8/19) for a long time (maybe before Cx was a thing?). So wondering if there are any backward compatibility issues? Like when replaying old transactions?

Reply:
should be fine. gencodec won't affect transaction data format.